src/main/java/org/folio/rest/impl/RequestObjectFactory.java

```diff
@@ -42,20 +41,21 @@
       });
   }
 
-  private CompletableFuture<RequestType> getRequestType(String patronId, String itemId, HttpClientInterface httpClient ) {
+  private CompletableFuture<RequestType> getRequestType(String patronId, String itemId) {
 
-    CompletableFuture<JsonObject> userFuture = LookupsUtils.getUser(patronId, okapiHeaders, httpClient);
-    CompletableFuture<JsonObject> itemFuture = LookupsUtils.getItem(itemId, okapiHeaders, httpClient);
+    CompletableFuture<JsonObject> userFuture = LookupsUtils.getUser(patronId, okapiHeaders);
+    CompletableFuture<JsonObject> itemFuture = LookupsUtils.getItem(itemId, okapiHeaders);
 
     RequestTypeParameters requestTypeParams = new RequestTypeParameters();
 
     return CompletableFuture.allOf(userFuture, itemFuture)
       .thenApply(x -> createRequestPolicyIdCriteria(itemFuture, userFuture, requestTypeParams))
-      .thenCompose((RequestTypeParameters criteria) -> lookupRequestPolicyId(criteria, httpClient))
+      .thenCompose(this::lookupRequestPolicyId)
       .thenCompose(policyIdResponse ->
-          LookupsUtils.getRequestPolicy(policyIdResponse.getString("requestPolicyId"), okapiHeaders, httpClient))
+          LookupsUtils.getRequestPolicy(policyIdResponse.getString("requestPolicyId"), okapiHeaders))
       .thenApply(RequestPolicy::from)
-      .thenApply(requestPolicy -> getRequestType(requestPolicy, requestTypeParams.getItemStatus()));
+      .thenApply(requestPolicy -> getRequestType(requestPolicy, requestTypeParams.getItemStatus()))
+      .exceptionally(t -> null);
```

Comment:
If the request completes with an exception, shouldn't we at least log a notice of the event to error?

Reply:
Ah, this was likely left in by mistake and should be removed :-/ 

I added `.exceptionally(t -> null)` to the end of future chains to allow for debugger break points. These weren't intended to be permanent.